### PR TITLE
VideoPress: set video block availability, based on site plan

### PR DESCRIPTION
**Pull request by retrofox**
This PR shows an upsell nudge in the VideoPress video block depending on the site platform (Jetpack, Atomic, Simple) and site plan.

Fixes partially https://github.com/Automattic/jetpack/issues/28330

## Proposed changes:

* VideoPress: set video block availability based on the site platform and plan

### Other information:

- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

### Show upsell nudge table

Platform | Simple | Atomic | Jetpack
------|------|-----|----
With free plan | Yes | Yes | No
With paid plan | No | No | No

* Go to the block editor
* Add a video block instance
* Depending on the site platform and the plan, confirm the upsell banner raises up when the user selects the block

image

### CODE REVIEW

**Comment by nunyvega** [comment]
Not working for me, this is what I got on a Free site (wpcom):
Screenshot 2023-01-17 at 10 52 02
The first image is a newly added block.
The second image is after trying to upload a video.

File: projects/plugins/jetpack/extensions/blocks/videopress/videopress.php [hunk 1]
```diff
@@ -41,19 +41,3 @@ function load_assets( $attrs, $content ) {
...
-// Set the videopress/video feature availability.
-add_action(
-	'jetpack_register_gutenberg_extensions',
-	function () {
-		\Jetpack_Gutenberg::set_extension_available( 'videopress/video' );
-	}
-);
-
-// Set the videopress/video-chapters feature availability.
-add_action(
```
Comment (lhkowalski) [inline projects/plugins/jetpack/extensions/blocks/videopress/videopress.php:54]: By removing this we are also unregistering the Video Chapters block, right?

**Comment by lhkowalski** [comment]
Tested on a Simple Site. With a free plan, I get this (correct, upgrade banner):

Screen Shot 2023-01-17 at 15 22 02

With a premium upgrade, I get this (correct, NO upgrade banner):

Screen Shot 2023-01-17 at 15 24 33

**Comment by lhkowalski** [comment]
Also tested on a Jetpack site. With a free plan (correct, NO upgrade banner):

Screen Shot 2023-01-17 at 15 31 48

With a monthly VideoPress plan, I get this (correct, NO upgrade banner):

Screen Shot 2023-01-17 at 15 41 58

File: projects/plugins/jetpack/extensions/blocks/videopress/videopress.php (hunk 1 above)
Comment (retrofox) [reply projects/plugins/jetpack/extensions/blocks/videopress/videopress.php:54]: yes, the Video Chapters block has been removed completely in fact

**Comment by lhkowalski** [comment]
I also tested on an atomic site, with the Business upgrade (correct, NO upgrade banner):

Screen Shot 2023-01-17 at 17 29 14

Now we still need to test using an atomic with a free plan (somehow). @retrofox gave the idea of downgrading the plan for the site using the store admin, I will try this approach.

**Comment by lhkowalski** [comment]
We explored the possibility of an Atomic site without a plan/with a free plan, and seems like it's not a real possibility for now. Atomic sites without a plan won't work, the `wp-admin` panel will be broken and the usage within Calypso will not be possible (the editor will not open, and the site resources will not be available).

This way, we can ignore the possibility of the scenario on the context of this change. Given we covered all the other scenarios, I think the change is working as expected:

Platform | Simple | Atomic | Jetpack
------|------|-----|----
With free plan | Showing -> Correct | Not possible currently | Not showing -> Correct
With paid plan | Not showing -> Correct | Not showing -> Correct | Not showing -> Correct

**Review by lhkowalski — APPROVED** [review]
Code change looks good.

**Pull request merged**